indexer/restriction_loader.cpp

```diff
@@ -0,0 +1,51 @@
+#include "indexer/restriction_loader.hpp"
+
+#include "coding/reader.hpp"
+
+using namespace routing;
+
+namespace
+{
+template <class TSource>
+void DeserializeRestrictions(Restriction::Type type, uint32_t count, TSource & src,
```

Comment:
Why not to just move all this code to the restriction serializer, and serialize a list of restrictions? It's quite strange that your serializer now is constructed from the current restriction, but serializes a previous restriction... a mess. Please, fix it.